themes/dotmatrix/dotmatrix.theme

```diff
@@ -0,0 +1,895 @@
+
+########################################################################
+# API
+########################################################################
+
+_lp_dotmatrix_theme_activate() {
+
+    LP_ENABLE_VCS_DIFFSTATS=${LP_ENABLE_UNCOMMITED:-0}
+
+    DOTMATRIX_DISPLAY_TEXT=${DOTMATRIX_DISPLAY_TEXT:-"T"}
+    DOTMATRIX_DISPLAY_X11=${DOTMATRIX_DISPLAY_X11:-"X"}
+    DOTMATRIX_HOST_LOCAL=${POWERLINE_HOST_LOCAL:-"⌂"}
+    DOTMATRIX_MARK_STASH=${DOTMATRIX_MARK_STASH:-"✝"}
+    DOTMATRIX_MARK_UNTRACKED=${DOTMATRIX_MARK_UNTRACKED:-"?"} # ⛶
+    DOTMATRIX_DIFF=${DOTMATRIX_DIFF:-"≠"}
+
+    # Sets of characters
+    DOTMATRIX_VARIANT=${DOTMATRIX_VARIANT:-"dotmatrix"}
+    if [[ "${DOTMATRIX_VARIANT}" == "dotmatrix" ]] ; then
+        DOTMATRIX_SPACED=${DOTMATRIX_SPACED:-0}
+
+        # Dotmatrix theme
+        DOTMATRIX_HOST_LOCAL=${POWERLINE_HOST_LOCAL:-"⌂"}
+        DOTMATRIX_USER_LOCAL=${DOTMATRIX_USER_LOCAL:-"♟"}
+        DOTMATRIX_VCS=${DOTMATRIX_VCS:-"╟┘"}
+        DOTMATRIX_MARK=${DOTMATRIX_MARK:-"■"}
+        DOTMATRIX_LINE=${DOTMATRIX_LINE:-"━"}
+        DOTMATRIX_PLUG=${DOTMATRIX_PLUG:-"▛"}
+        DOTMATRIX_SOFT=${DOTMATRIX_SOFT:-"╱"}
+        DOTMATRIX_MEAN=${DOTMATRIX_MEAN:-"▞"}
+        # Separator arrays                                (left center right 
+        [[ -z ${DOTMATRIX_LINK[@]+x} ]] && DOTMATRIX_LINK=("▙" " "   "▜")
```

Comment:
I know it's verbose, but array checking needs to be done with `${array[@]+"${array[@]}"}` to support `set -u`. See [the wiki](https://github.com/nojhan/liquidprompt/wiki/Shell-standards#arrays). And if you are checking for a not set array to set a default, it needs to be this more complex form.

```suggestion
        DOTMATRIX_LINK=( ${DOTMATRIX_LINK[@]+"${DOTMATRIX_LINK[@]}"} )
        [[ ${#DOTMATRIX_LINK[@]} == 0 ]] && DOTMATRIX_LINK=("▙" " "   "▜")
```